test_crates/repr_c_plain_struct_fields_reordered/new/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+// Public repr(C) struct with reordered fields - should trigger warning
+#[repr(C)]
+pub struct PublicStruct {
+    pub b: u16, // moved
+    pub c: u32, // moved
+    pub a: u8,  // moved
+}
+
+// Private repr(C) struct with reordered fields - should not trigger
+#[repr(C)]
+struct PrivateStruct {
+    pub b: u16,
+    pub c: u32,
+    pub a: u8,
+}
+
+// Public non-repr(C) struct with reordered fields - should not trigger
+pub struct RegularStruct {
+    pub b: u16,
+    pub c: u32,
+    pub a: u8,
+}
+
+// Public repr(C) struct with doc(hidden) - should not trigger
+#[doc(hidden)]
```

Comment:
Oh this is a good idea to test, and gave me an idea for another thing to check: what if the `pub` field itself is `doc(hidden)`?

I think our answer should be similar to the private fields below: we don't trigger on that field, but if it affects the order of anything else then we trigger the lint. If only `doc(hidden)` fields get reordered, then the lint doesn't trigger at all.